### ISSUE TITLE
[bug] When IdentityClients set more then one(Default),HttpClientAuthenticator will fail

### ISSUE THREAD

**Issue by yinchang0626** [abpframework/abp#9102]
* Your **ABP Framework version**.
4.3.0
* Your **User Interface** type (Angular/MVC/React... etc.) if the issue is related to a specific UI
don't care
* Your database provider(EF Core/MongoDB)
don't care
* Exception message and **stack trace** if available (check the logs).
`Volo.Abp.Http.Client.AbpRemoteCallException: 'Unauthorized'`
* Steps needed to **reproduce** the problem.
For test this problem,I use `MyCompanyName.MyProjectName.HttpApi.Client.ConsoleTestApp` in `module` of `template`:
1:`update-database` for `MyCompanyName.MyProjectName.IdentityServer`
2.Run `MyCompanyName.MyProjectName.HttpApi.Host` and `MyCompanyName.MyProjectName.IdentityServer`
3.modify `appsettings.json` in `MyCompanyName.MyProjectName.HttpApi.Client.ConsoleTestApp`like:
```json
{
  "RemoteServices": {
    "Default": {
      "BaseUrl": "https://localhost:44301/"
    },
    "MyProjectName": {
      "BaseUrl": "https://localhost:44300/"
    }
  },
  "IdentityClients": {
    "MyProjectName": {
      "GrantType": "password",
      "ClientId": "MyProjectName_App",
      "ClientSecret": "1q2w3e*",
      "UserName": "admin",
      "UserPassword": "1q2w3E*",
      "Authority": "https://localhost:44301/",
      "Scope": "MyProjectName"
    }
  }
}
```
4.Run `MyCompanyName.MyProjectName.HttpApi.Client.ConsoleTestApp` will receive error
![image](https://user-images.githubusercontent.com/16490084/118972102-c112b000-b9a2-11eb-9eb1-120ee4ad6b37.png)
5.After check source code,I think something wrong with `IdentityModelRemoteServiceHttpClientAuthenticator` 
```csharp
 public class IdentityModelRemoteServiceHttpClientAuthenticator : IRemoteServiceHttpClientAuthenticator, ITransientDependency
    {
        protected IIdentityModelAuthenticationService IdentityModelAuthenticationService { get; }

        public IdentityModelRemoteServiceHttpClientAuthenticator(
            IIdentityModelAuthenticationService identityModelAuthenticationService)
        {
            IdentityModelAuthenticationService = identityModelAuthenticationService;
        }

        public virtual async Task Authenticate(RemoteServiceHttpClientAuthenticateContext context)
        {
            await IdentityModelAuthenticationService.TryAuthenticateAsync(
                context.Client,
                context.RemoteService.GetIdentityClient()//context.RemoteServiceName
            );
        }
    }
```
If I modify `context.RemoteService.GetIdentityClient()` to `context.RemoteServiceName`,It's working fine
Could you check it ,thanks

**Comment by Loongle** [comment]
What is the temporary fix? @yinchang0626

**Comment by yinchang0626** [comment]
@Loongle  Hi
 In your `HttpApi.Host` add a class,
```csharp
    [Dependency(ReplaceServices = true)]
    public class MyIdentityModelRemoteServiceHttpClientAuthenticator : IdentityModelRemoteServiceHttpClientAuthenticator, IRemoteServiceHttpClientAuthenticator, ITransientDependency
    {
        public MyIdentityModelRemoteServiceHttpClientAuthenticator(
            IIdentityModelAuthenticationService identityModelAuthenticationService)
            : base(identityModelAuthenticationService)
        {
        }

        public override async Task Authenticate(RemoteServiceHttpClientAuthenticateContext context)
        {
            await IdentityModelAuthenticationService.TryAuthenticateAsync(
                context.Client,
                context.RemoteServiceName
            );
        }
    }
```

**Comment by yinchang0626** [comment]
resolved by #9151
